### PR TITLE
WebManager: Map loading changes

### PR DESCRIPTION
**Pull request by kristofbolyai**
This PR adds more logic, replaces map list with a much faster concurrent object and tries to remove the "hacky" parts of map piece downloading.

### CODE REVIEW

**Comment by kristofbolyai** [comment]
@Incompleteusern can you confirm these changes are okay?

File: common/src/main/java/com/wynntils/features/user/overlays/map/MiniMapOverlayFeature.java [hunk 1]
```diff
@@ -107,7 +107,7 @@ public void render(PoseStack poseStack, float partialTicks, Window window) {
 
             // TODO replace with generalized maps whenever that is done
             if (WebManager.isMapLoaded()) {
-                MapProfile map = WebManager.getMaps().get(0);
+                MapProfile map = WebManager.getMaps().peek();
```
Comment (HighCrit) [inline common/src/main/java/com/wynntils/features/user/overlays/map/MiniMapOverlayFeature.java:110]: Why no .poll() isnt or do we need to contents in the future elsewhere?

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/user/overlays/map/MiniMapOverlayFeature.java:110]: This is render loop. We need this every render.

File: common/src/main/java/com/wynntils/core/webapi/WebManager.java [hunk 2]
```diff
@@ -286,34 +295,35 @@ public static CompletableFuture<Boolean> tryLoadMaps() {
...
-                        }
-                    });
+                    if (downloadedMapProfiles.size() == mapArray.size()) {
+                        maps = downloadedMapProfiles;
+                        result.complete(true);
+                    } else {
+                        WynntilsMod.error("Expected to have" + mapArray.size() + " map parts, got "
+                                + downloadedMapProfiles.size() + " parts.");
+                        result.complete(false);
+                    }
```
Comment (Incompleteusern) [inline common/src/main/java/com/wynntils/core/webapi/WebManager.java:326]: The reason for the hack is that having added a bunch of map download requests done asynchronously, we need to do an operation when all are complete.

Now, we are doing the whole map download synchronously, which likely dramatically increases startup time

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/core/webapi/WebManager.java:326]: Yeah, but can't that make a race somewhere? As we do not group requests, and only dispatch 1 by 1

**Comment by kristofbolyai** [comment]
@Incompleteusern are you fine with these changes, or do you think we should close this?

File: common/src/main/java/com/wynntils/core/webapi/WebManager.java (hunk 2 above)
Comment (Incompleteusern) [reply common/src/main/java/com/wynntils/core/webapi/WebManager.java:326]: I feel the solution here is to add better grouping functionality/use the existing grouping functionality and code that runs when the entire group is done by default.

**Comment by Incompleteusern** [comment]
I would rather we keep the async functionality and the hack than remove the hack.

**Comment by Incompleteusern** [comment]
I may try my own hand at removing the hack sometime.

**Pull request closed without merging**